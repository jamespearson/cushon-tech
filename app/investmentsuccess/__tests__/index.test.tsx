import { render, screen, fireEvent } from '@testing-library/react-native'
import { router } from 'expo-router'
import InvestmentSuccessScreen from '..'

jest.mock('expo-router', () => ({
  router: {
    push: jest.fn()
  }
}))

describe('Investment Success Screen', () => {
  it('should render as expected', () => {
    render(<InvestmentSuccessScreen />)

    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should navigate to the Portfoilo screen', () => {
    render(<InvestmentSuccessScreen />)

    fireEvent.press(screen.getByText('VIEW INVESTMENTS'))

    expect(router.push).toHaveBeenCalledWith('portfolio')
  })
})
