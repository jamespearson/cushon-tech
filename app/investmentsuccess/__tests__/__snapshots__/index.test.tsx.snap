// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Investment Success Screen should render as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "gap": 24,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <Text />
  <Text
    style={
      {
        "fontFamily": "MuseoSansRounded500",
        "fontSize": 32,
        "lineHeight": 42,
      }
    }
  >
    Congratulations
  </Text>
  <Text
    style={
      {
        "fontFamily": "MuseoSansRounded700",
        "fontSize": 18,
        "lineHeight": 24,
      }
    }
  >
    Your investment was successful.
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "stretch",
        "backgroundColor": "#df2f89",
        "borderColor": "#df2f89",
        "borderRadius": 8,
        "borderWidth": 1,
        "padding": 24,
      }
    }
  >
    <View
      style={{}}
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "MuseoSansRounded700",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        VIEW INVESTMENTS
      </Text>
    </View>
  </View>
</View>
`;
